upgraderesponder/service.go

```diff
@@ -382,41 +470,63 @@ func (s *Server) recordRequest(httpReq *http.Request, req *CheckUpgradeRequest)
 	}
 
 	// We use IP to find the location but we don't store IP
-	loc, err := s.getLocation(publicIP)
+	location, err := s.getLocation(publicIP)
 	if err != nil {
 		logrus.Error("Failed to get location for one ip")
 	}
 
 	if s.influxClient != nil {
-		var (
-			err error
-			pt  *influxcli.Point
-		)
+		var err error
 		defer func() {
 			if err != nil {
 				logrus.Errorf("Failed to recordRequest: %v", err)
 			}
 		}()
 
-		tags := map[string]string{
-			InfluxDBTagAppVersion: req.AppVersion,
-		}
-		for k, v := range req.ExtraInfo {
-			tags[utils.ToSnakeCase(k)] = v
-		}
-		fields := map[string]interface{}{
-			utils.ToSnakeCase(ValueFieldKey): ValueFieldValue,
-		}
-		if loc != nil {
-			tags[InfluxDBTagLocationCity] = loc.City
-			tags[InfluxDBTagLocationCountry] = loc.Country.Name
-			tags[InfluxDBTagLocationCountryISOCode] = loc.Country.ISOCode
-		}
-		pt, err = influxcli.NewPoint(InfluxDBMeasurement, tags, fields, time.Now())
-		if err != nil {
-			return
-		}
+		s.recordTagsFromRequest(req, location)
+		s.recordFieldsFromRequest(req)
+	}
+}
 
-		s.dbCache.AddPoint(pt)
+func (s *Server) recordTagsFromRequest(req *CheckUpgradeRequest, location *Location) {
+	tags := map[string]string{
+		InfluxDBTagAppVersion: req.AppVersion,
 	}
+	extraTagInfo := utils.MergeStringMaps(req.ExtraInfo, req.ExtraTagInfo)
+	for k, v := range extraTagInfo {
+		tags[utils.ToSnakeCase(k)] = v
+	}
+
+	if location != nil {
+		tags[InfluxDBTagLocationCity] = location.City
+		tags[InfluxDBTagLocationCountry] = location.Country.Name
+		tags[InfluxDBTagLocationCountryISOCode] = location.Country.ISOCode
+	}
+
+	fields := map[string]interface{}{
+		utils.ToSnakeCase(ValueFieldKey): ValueFieldValue,
+	}
+
+	pt, err := influxcli.NewPoint(InfluxDBMeasurement, tags, fields, time.Now())
+	if err != nil {
+		logrus.WithError(err).Error("Failed to record tags from request")
+		return
+	}
+
+	s.dbCache.AddPoint(pt)
+}
+
+func (s *Server) recordFieldsFromRequest(req *CheckUpgradeRequest) {
+	fields := make(map[string]interface{}, len(req.ExtraFieldInfo))
+	for k, v := range req.ExtraFieldInfo {
+		fields[utils.ToSnakeCase(k)] = v
+	}
+
+	pt, err := influxcli.NewPoint(InfluxDBMeasurement, nil, fields, time.Now())
+	if err != nil {
+		logrus.WithError(err).Error("Failed to record fields from request")
+		return
+	}
+
+	s.dbCache.AddPoint(pt)
```

Comment:
We add 2 points for 1 request which will make the number of node count double because the number of node == number of request